dts/bindings/pinctrl/gd,gd32-pinctrl-common.yaml

```diff
@@ -0,0 +1,26 @@
+# Copyright (c) 2021 Teslabs Engineering S.L.
+# SPDX-License-Identifier: Apache-2.0
+
+include:
+    - name: base.yaml
+    - name: pincfg-node-group.yaml
+      child-binding:
+        child-binding:
+          property-allowlist:
+            - drive-push-pull
+            - drive-open-drain
+            - bias-disable
+            - bias-pull-down
+            - bias-pull-up
```

Comment:
bias-high-impedance is missing

Reply:
This option is not required, same as STM32 (see https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/bindings/pinctrl/st%2Cstm32-pinctrl.yaml#L18). Not setting any bias and operating as input would be sufficient as far as I understand.

Reply:
> Not setting any bias and operating as input would be sufficient as far as I understand.

I understand not setting any bias, which means `bias-disable` (default state), equals `Analog`. We should not work with a hidden state because people can made confusion. The configuration table have 4 options for Input and there are 3 options defined. If you look at https://github.com/zephyrproject-rtos/zephyr/blob/a5c2a6169b65849b48e3416c16b786aa53078562/dts/bindings/pinctrl/pincfg-node.yaml#L25-L28 this is the option that Linux define for the `Input Floating`.

Reply:
The problem is that analog is treated via the pinmux value as a separate mode, not as an input mode. In case of DAC (an output signal), ANALOG still has to be used. For low-power mode, what is usually done is to set the analog mode (default config) on a pin, this is why I added the analog option for all pins in the generator:

```
/* ANALOG */
#define ANALOG_PA0 \
	GD32_PINMUX_AFIO('A', 0, ANALOG, NORMP)
#define ANALOG_PA1 \
	GD32_PINMUX_AFIO('A', 1, ANALOG, NORMP)
...
```

I know it can be a bit confusing, it's really STM32 inheritance (it's done this way, see for example https://elixir.bootlin.com/linux/latest/source/arch/arm/boot/dts/stm32mp15-pinctrl.dtsi#L62). I think that treating analog in a special way makes things more understandable when it comes to the analog signals, but not that clear for low-power modes. In such case, it would probably make more sense to configure the pin as an input and set bias-high-impedance.

Reply:
In this case, it is important give some background for users. The information you presented at last comment is relevant to users understand how this should be configured. Other important thing is that no one can determine from where user will start to read the documentation. I think it could be very valuable improve documentation and try connect those things. For instance, correlate the _GPIO configuration table_ with the options available can be very valuable to users. The same way, explain to then the why Analog is configured differently will help. Maybe provide pointer that exists in Zephyr can help to. Take in consideration that a lot of users don't know Linux and devicetree.

Reply:
I have updated the bindings description explaining more details on low power modes, please check.